tslib/src/functions/check-notifier/index.ts

```diff
@@ -0,0 +1,29 @@
+import { AzureFunction, Context } from '@azure/functions'
+import { performance } from 'perf_hooks'
+import { ICheckNotificationMessage } from './check-notification-message'
+import { CheckNotifier } from './check-notifier.v1'
+const functionName = 'check-notifier'
+
+const serviceBusTrigger: AzureFunction = async function (context: Context, checkNotification: ICheckNotificationMessage): Promise<void> {
+  const start = performance.now()
+  const version = checkNotification.version
+  context.log.info(`${functionName}: version:${version} message received for checkCode ${checkNotification.checkCode}`)
+  try {
+    if (version !== 1) {
+      // dead letter the message as currently we only support v1
+      throw new Error(`Message schema version:${version} unsupported`)
```

Comment:
If this error is not caught the message will be re-injected into the queue after a delay, again and again until the limit is reached.

If we don't want to re-process this, we should silently swallow the message by just returning.  It would be worthwhile logging a message too.

Reply:
good idea. will review

Reply:
in order for it to end up on the poison queue for later inspection i have retained the error throw, but also added logging.